src/enable_variable_ops/ngraph_encapsulate_op.cc

```diff
@@ -0,0 +1,905 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#include <cstdlib>
+#include <mutex>
+#include <utility>
+
+#include "tensorflow/core/common_runtime/dma_helper.h"
+#include "tensorflow/core/common_runtime/optimization_registry.h"
+#include "tensorflow/core/framework/graph.pb.h"
+#include "tensorflow/core/framework/node_def_util.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/graph/graph_constructor.h"
+
+#include "ngraph_backend_manager.h"
+#include "ngraph_builder.h"
+#include "ngraph_catalog.h"
+#include "ngraph_cluster_manager.h"
+#include "ngraph_freshness_tracker.h"
+#include "ngraph_log.h"
+#include "ngraph_mark_for_clustering.h"
+#include "ngraph_timer.h"
+#include "ngraph_utils.h"
+#include "ngraph_var.h"
+
+#include "ngraph/event_tracing.hpp"
+#include "ngraph/runtime/backend.hpp"
+
+#if defined NGRAPH_DISTRIBUTED
+#include "ngraph/distributed.hpp"
+#endif
+
+using namespace std;
+namespace ng = ngraph;
+
```

Comment:
Maybe we should add comments at the beginning here on what changed in this file compared to https://github.com/NervanaSystems/ngraph-tf/blob/master/src/ngraph_encapsulate_op.cc, like how you added for `ngraph_tracked_variable.cc`